week-four/app/src/main/java/pro/marvinhosea/movielist/ui/login/UserLoginActivity.kt

```diff
@@ -0,0 +1,66 @@
+package pro.marvinhosea.movielist.ui.login
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.EditText
+import android.widget.Toast
+import pro.marvinhosea.movielist.R
+import pro.marvinhosea.movielist.repository.UserSharedPrefRepository
+import pro.marvinhosea.movielist.ui.movies.MainActivity
+
+class UserLoginActivity : AppCompatActivity() {
+
+    private val repository = UserSharedPrefRepository
```

Comment:
If you're using a singleton, it doesn't make sense to add it to a property. That doesn't do anything.

Choose one of the two - singleton, or non-singleton and property initialization.